assets/app/scripts/services/auth.js

```diff
@@ -276,4 +276,18 @@ angular.module('openshiftConsole')
       }
     }
   };
-}]);
+}])
+
+/* A WebSocket factory for kubernetesContainerTerminal */
+.factory("AuthContainerWebSocket", function(API_CFG, AuthService) {
```

Comment:
Also, `$ws` should be updated to plumb the optional second `protocols` param to `new WebSocket`